pkgs/by-name/gt/gtk-session-lock/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, meson
+, ninja
+, pkg-config
+, gobject-introspection
+, vala
+, gtk3
+, wayland-scanner
+, wayland
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gtk-session-lock";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "Cu3PO42";
+    repo = "gtk-session-lock";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-SHKAYmdev08oRB/V6UpfSFqYwplF59IaNSOoWcACPig=";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    gobject-introspection
+    vala
+  ];
+
+  buildInputs = [
+    gtk3
+    wayland-scanner
```

Comment:
Doesn't Wayland-scanner go to nativeBuildInputs?

Reply:
I've set `strictDeps`, if it belongs to `nativeBuildInputs` it should fail?

Reply:
Yeah, it should but in most places it is there anyway. Maybe buildRustProject is not fully correct about it?

Reply:
I was a little skeptical the last time I saw this suggestion, I'll look into it later today.